src/WebJobs.Extensions.SendGrid/Config/SendGridJobHostConfigurationExtensions.cs

```diff
@@ -48,7 +54,26 @@ public void Initialize(ExtensionConfigContext context)
                     throw new ArgumentNullException("context");
                 }
 
-                context.Config.RegisterBindingExtension(new SendGridAttributeBindingProvider(_sendGridConfig, context.Config.NameResolver));
+                if (string.IsNullOrEmpty(_sendGridConfig.ApiKey))
+                {
+                    throw new InvalidOperationException(
+                        string.Format("The SendGrid ApiKey must be set either via a '{0}' app setting, via a '{0}' environment variable, or directly in code via SendGridConfiguration.ApiKey.",
+                        SendGridConfiguration.AzureWebJobsSendGridApiKeyName));
+                }
+                _sendGrid = new Web(_sendGridConfig.ApiKey);
+
+                IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+                converterManager.AddConverter<JObject, SendGridMessage>(SendGridHelpers.CreateMessage);
```

Comment:
This JObject->SendGridMessage converter is the key that allows this binding to work for Node and other languages.